doc/sphinx/proof-engine/tactics.rst

```diff
@@ -105,6 +105,227 @@ bindings_list`` where ``bindings_list`` may be of two different forms:
   .. exn:: Not the right number of missing arguments.
      :undocumented:
 
+.. _intropatterns:
+
+Intro patterns
+~~~~~~~~~~~~~~
+
+Intro patterns let you specify the name to assign to variables and hypotheses
+introduced by tactics.  They also let you split an introduced hypothesis into
+multiple hypotheses or subgoals.  Common tactics that accept intro patterns
+include :tacn:`intros`, :tacn:`destruct` and :tacn:`apply`.
+
+.. productionlist::        coq
+   simple_intropattern     : `simple_intropattern_closed`
+                           : % `operconstr` ... % `operconstr`
+   simple_intropattern_closed : `or_and_intropattern`
+                              : `equality_intropattern`
+                              : _
+                              : `naming_intropattern`
+   or_and_intropattern     : [ `intropatterns` | ... | `intropatterns` ]
+                           : ()
+                           : ( `simple_intropattern` , ... , `simple_intropattern` )
+                           : ( `simple_intropattern` & ... & `simple_intropattern` )
+   equality_intropattern   : ->
+                           : <-
+                           : [= `intropatterns` ]
+   intropatterns           : `nonsimple_intropattern` ...
+                           : <empty>
+   ne_intropatterns        : `nonsimple_intropattern` ...
+   nonsimple_intropattern  : `simple_intropattern`
+                           : *
+                           : **
+   naming_intropattern     : `pattern_ident`
+                           : ?
+                           : `ident`
+   pattern_ident           : ? `ident`
+
+Use these elementary patterns to specify a name:
+
+* :n:`@ident` - use the specified name
+* :n:`?` - let Coq choose a name
+* :n:`? @ident` - generate a name that begins with :n:`@ident`
```

Comment:
```suggestion
* :n:`?@ident` - generate a name that begins with :n:`@ident`
```

There should be no space, otherwise it will be parsed as two intropatterns. I suppose the grammar should be fixed as well to reflect that.